src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java

```diff
@@ -167,17 +168,46 @@ public String key() {
     long updatedPlayheadPosition;
     /** The total time in seconds a video has been in a paused state during a video session */
     long offset = 0;
-
+    /** Whether the video playhead is in a paused state. */
     boolean isPaused = false;
+    /** Video bitrate; set or updated whenever {@link #setQosVariables(Properties)} is invoked */
+    long bitrate;
+    /**
+     * Video startup time; set or updated whenever {@link #setQosVariables(Properties)} is invoked
+     */
+    double startupTime;
+    /**
+     * Video frames per second; set or updated whenever {@link #setQosVariables(Properties)} is
+     * invoked
+     */
+    double fps;
+    /**
+     * Number of dropped frames during video playback; set or updated whenever {@link
+     * #setQosVariables(Properties)} is invoked
+     */
+    long droppedFrames;
 
     public PlaybackDelegate() {
       this.initialTime = System.currentTimeMillis();
     }
 
-    /** TO DO: Implement quality of service method. */
+    /**
+     * Sets the value of member variables related to quality of service tracking.
+     *
+     * @param properties Properties object from a "Video Quality Updated" event, which triggers
+     *     invocation of this method.
+     */
+    public void setQosVariables(Properties properties) {
+      this.bitrate = properties.getLong("bitrate", 0);
```

Comment:
If the customer sends all these fields, some will get set to zero. If this is intentional that's ok too, in that case we might as well create the QOS media object here and store a reference to it that is returned in the other method. Otherwise we're unnecessarily creating objects even when nothing changed.

Reply:
Though given that this is every 10 seconds and not every second like a thought, this doesn't need to be changed, though it will simplify our book keeping since we'll only need to track one field.